src/server.ts

```diff
@@ -1,21 +1,90 @@
+import bodyParser from "body-parser";
+import cors from "cors";
 import "dotenv/config";
 import express from "express";
+import session from "express-session";
+import validator from "validator";
+import logger from "./loggingFramework/logger";
+import { formBasedAuth } from "./middleware/formBasedAuthentication";
 
 const app = express();
-const port = process.env.PORT || (process.env.NODE_ENV === 'test' ? 4000 : 3000);
+
+if (!process.env.SESSION_SECRET_KEY) {
```

Comment:
Let's have a variable like `const sessionSecretKey = process.env.SESSION_SECRET_KEY` and use it in the two places you have used it.